lib/facade_converter.ts

```diff
@@ -92,19 +92,33 @@ export class FacadeConverter extends base.TranspilerBase {
    * Searches for type references that require extra imports and emits the imports as necessary.
    */
   emitExtraImports(sourceFile: ts.SourceFile) {
-    var hasXmlHttpRequest = (n: ts.Node): boolean => {
-      if (n.kind === ts.SyntaxKind.TypeReference) {
-        var type = base.ident((<ts.TypeReferenceNode>n).typeName);
-        if (type === 'XMLHttpRequest') {
-          return true;
+    var libraries = <ts.Map<string>>{
+      "XMLHttpRequest": "dart:html",
+      "Uint8Array": "dart:typed_arrays",
+      "ArrayBuffer": "dart:typed_arrays"
+    };
+
+    for (var typeName in libraries) {
+      if (libraries.hasOwnProperty(typeName)) {
+        var library = libraries[typeName];
+        if (this._hasType(sourceFile, typeName)) {
+          this.emit(`import "${library}";`);
         }
       }
-      return n.getChildren(sourceFile).some(hasXmlHttpRequest);
-    };
+    }
+  }
 
-    if (hasXmlHttpRequest(sourceFile)) {
-      this.emit('import "dart:html";');
+  private _hasType(n: ts.Node, typeName: String, sourceFile?: ts.SourceFile): boolean {
+    if (n.kind === ts.SyntaxKind.TypeReference) {
```

Comment:
This walks the entire AST for a source file n times for each of the - currently three - types that need special casing. Can we do this in just one walk, e.g. by keeping track of what extra imports we emitted already, and then emitting directly during the search?